app/controllers/vinyls_controller.rb

```diff
@@ -4,20 +4,23 @@ def index
   end
 
   def show
-    @vinyl = Vinyl.find(params[:id])
+    @user = User.find(params[:user_id])
+    @vinyl = @user.vinyls
     @reviews = @vinyl.reviews
-    @review = @vinyl.reviews.build
+    @review = @vinly.reviews.build
   end
 
   def new
+    @user = User.find(params[:user_id])
     @vinyl = Vinyl.new
   end
 
   def create
     @vinyl = Vinyl.new(params.require(:vinyl).permit(:title, :artist, :year, :label))
```

Comment:
require(:vinyl).permit(:title, :artist, :year, :label) put this in a private method liek we've been doing